controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -411,12 +502,27 @@ func (sp *servicePublisher) newPortPublisher(srcPort Port, hostname string) *por
 		metrics:    endpointsVecs.newEndpointsMetrics(sp.metricsLabels(srcPort, hostname)),
 	}
 
-	endpoints, err := sp.k8sAPI.Endpoint().Lister().Endpoints(sp.id.Namespace).Get(sp.id.Name)
-	if err != nil && !apierrors.IsNotFound(err) {
-		sp.log.Errorf("error getting endpoints: %s", err)
-	}
-	if err == nil {
-		port.updateEndpoints(endpoints)
+	if !sp.k8sAPI.EndpointSliceAccess() {
```

Comment:
nit: I'd flip the branches of this if-else to get rid of the negation in the condition.

Reply:
this fixed itself (in a way) after I changed the return type for `EndpointSliceAccess`